### PR TITLE
KEYCLOAK-14234 KEYCLOAK-13666 support for running adapter tests with different hostnames for auth and app servers

### PR DESCRIPTION
**Pull request by mhajas**
This is quite an extensive change. It makes the testsuite running with two separate hosts (for app and auth server) and also it fixes some issues with enabling SSL for app server. The main reason is to be able to properly test the newest changes to the SameSite cookie attribute.

To test the changes run any adapter test for example with: 
```
-Dauth.server.host=auth-server.127.0.0.1.nip.io -Dapp.server.host=app-server.127.0.0.1.xip.io -Dapp.server.ssl.required=true
```

After all reviews, I will squash the commits together. For reviewing it will be probably better to have it in more commits.

Results from Jenkins:
Adapter: https://keycloak-jenkins.com/job/universal-test-pipeline-adapters/1534/
Server: https://keycloak-jenkins.com/job/universal-test-pipeline-server/1937/

### CODE REVIEW

**Review by vmuzikar — CHANGES_REQUESTED** [review]
Thanks a lot @mhajas! Nice work!

Added a few minor comments.

Looking at the test results, it seems like Windows 2012 is broken by this PR both in server and adapter test pipelines.

**Comment by vmuzikar** [comment]
It seems like the errors on Win 2012 were caused by too long working path. The path is derived from KC version and that is derived from branch name which is quite long in this case.
I explicitly set KC version to shorter name and reran the adapter tests: https://keycloak-jenkins.com/job/universal-test-pipeline-adapters/1547/.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/example/cors/CorsExampleAdapterTest.java [hunk 1]
```diff
@@ -190,7 +184,6 @@ private String getAuthServerVersion() {
                 "/auth/admin/master/console/#/server-info");
         jsDriverTestRealmLoginPage.form().login("admin", "admin");
 
-        WaitUtils.waitUntilElement(By.tagName("body")).is().visible();
```
[the remark this answers is not in the document]
Comment (mhajas) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/example/cors/CorsExampleAdapterTest.java:193]: I removed it because the `login` method is using `clickLink()` function to log in, so any wait after this should not be necessary. But I can revert it if you want.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/SAMLSameSiteTest.java [hunk 2]
```diff
@@ -38,9 +39,10 @@
 @AppServerContainer(ContainerConstants.APP_SERVER_TOMCAT9)
 @AuthServerContainerExclude(AuthServerContainerExclude.AuthServer.REMOTE)
 public class SAMLSameSiteTest extends AbstractSAMLServletAdapterTest {
```
[the remark this answers is not in the document]
Comment (mhajas) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/SAMLSameSiteTest.java:41]: I was thinking about this, but I was not sure. This test is running with different host even in the case when no app.server.host is configured. But you are probably right in this case, it is not useful at all, as in pipeline this test will be running with htmlunit, not chrome, and the auth server is running on `localhost`. So I will remove it.

**Comment by mhajas** [comment]
@vmuzikar I have run the pipeline again, and it seems the results are ok. https://keycloak-jenkins.com/job/universal-test-pipeline-adapters/1560/. I also fixed your suggestions except one with removing wait from the CORS test (I can revert if you want). Otherwise, it should be ready to merge.

File: testsuite/integration-arquillian/tests/base/pom.xml [hunk 3]
```diff
@@ -982,7 +982,7 @@
                                     <configuration>
                                         <keystore>${app.server.keystore}</keystore>
                                         <storepass>${app.server.keystore.password}</storepass>
-                                        <alias>localhost</alias>
+                                        <alias>localhost3</alias>
```
Comment (vramik) [inline testsuite/integration-arquillian/tests/base/pom.xml:985]: @mhajas, can you please clear for me why this change is important?

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/example/cors/CorsExampleAdapterTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (vmuzikar) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/example/cors/CorsExampleAdapterTest.java:193]: Thanks for the explanation. No need to revert. ;)

**Review by vmuzikar — DISMISSED** [review]
LGTM, thanks again @mhajas!

**Review by vramik — COMMENTED** [review]
Overall changes looks good to me. I haven't found anything what doesn't make sense. I have just two minor things (questions). Thanks @mhajas.

**Comment by mhajas** [comment]
Thank you very much for reviews @vmuzikar @vramik! I think all suggestions should be incorporated, so I squashed commits into one. This should be ready to merge

**Review by vmuzikar — APPROVED** [review]

**Review by vramik — APPROVED** [review]

**Review by hmlnarik — APPROVED** [review]
Approving, based on @vramik and @vmuzikar's approval. Thank you!
